src/progressive-profiling/ProgressiveProfiling.jsx

```diff
@@ -103,8 +109,12 @@ const ProgressiveProfiling = (props) => {
     return null;
   }
 
-  const optionalFields = location.state.optionalFields.fields;
-  const extendedProfile = location.state.optionalFields.extended_profile;
+  let optionalFields = {};
+  let extendedProfile = [];
+  if (location?.state) {
+    optionalFields = location.state.optionalFields.fields;
+    extendedProfile = location.state.optionalFields.extended_profile;
+  }
```

Comment:
why are you doing this? that's not the part of this ticket.

Reply:
By registering through the non-embedded experience we set this location state and use this state for progressive profiling, but through iframe embedded experience we are unable to set the location state that's why we need to handle this case when we are not setting the state.

Reply:
This will be handled in my PR.

Reply:
no.